### DISCUSSION TITLE
How to make nx test use the firebase emulator

### DISCUSSION THREAD

**Discussion by leon** [nrwl/nx#8031] [General]
Hi!,

We are using firebase firestore / storage / pubsub in our monorepo.

When we do our testing we want to run the firebase emulator.

At the moment we are using an npm script
```
"test": "TZ=utc firebase emulators:exec --only firestore 'nx run-many --target=test --all'"
```
which boots the emulator before running the jest tests.

The problem is that if you want to run a single test you usually run it against the project

```
nx test shared-data-access
```

but since we need to have the emulator running, you then have to wrap it in the above.

The question
### Can I somehow force NX to always start the emulator before running tests?

so that it doesn't matter if I use the `nx run-many` or `nx test` or in the ci using `nx affected:test`
it should always have the emulator available.

**Comment by mry-bigweb** [comment]
Hi @leon 

Did you ever find a solution for this? :)

**Reply by mry-bigweb** [reply]
Simple solution that seems to work:
```
"test-jest": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/apps/main-api"],
      "options": {
        "jestConfig": "apps/main-api/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "test": {
      "executor": "nx:run-commands",
      "options": {
        "command": "firebase emulators:exec 'nx run main-api:test-jest'"
      }
    }
```
